manager/controlapi/node.go

```diff
@@ -220,14 +240,19 @@ func (s *Server) UpdateNode(ctx context.Context, request *api.UpdateNodeRequest)
 	}
 
 	if demote && s.raft != nil {
-		memberlist := s.raft.GetMemberlist()
-		for raftID, member := range memberlist {
-			if member.NodeID == request.NodeID {
-				if err := s.raft.RemoveMember(ctx, raftID); err != nil {
-					return nil, err
+		if err := s.raft.RemoveMember(ctx, member.RaftID); err != nil {
+			// Rollback to the initial Spec if we can't finalize the role
+			// change by removing the member from raft.
+			s.store.Update(func(tx store.Tx) error {
+				node = store.GetNode(tx, request.NodeID)
+				if node == nil {
+					return nil
 				}
-				break
-			}
+
+				node.Spec = initSpec
```

Comment:
@abronan I think the notion that we have to rollback the spec is probably incorrect. The spec reflects the desired state. If the desired state cannot be achieved, we report on that and keep trying.

Is the issue that we are querying the spec for the current role, rather than gating it through a node field?